src/Imath/ImathRandom.cpp

```diff
@@ -11,7 +11,7 @@
 //-----------------------------------------------------------------------------
 
 #include "ImathRandom.h"
-#include "ImathInt64.h"
+#include <cstdint>
```

Comment:
Note that this line changing breaks source compatibility with other packages -- if they didn't redundantly include "ImathInt64.h", then now they will ONLY see cstdint, and thus not get the typedefs (the ones that are marked as deprecated, but are still present). This may be a bit of a more hard fail than we intended?

Reply:
I figured this out only when OIIO nightly CI against Imath master failed today. :-)

Reply:
@lgritz presumably you were using Int64 in OIIO, so got the fail?
I suppose leaving the includes of ImathInt64.h in place, but removing references to the types, would prevent compile errors, but still emit deprecation warnings if Int64/SInt64 were used. That seems like a more gentle approach.

A later release should remove the include statements, which would generate compile errors if the types were used without directly including ImathInt64. At that stage a #warning can be added to ImathInt64.h, that would be emitted whenever it is included, to make it easier to find where the include statements are.

Reply:
Yes, I was using Int64. So I expected the deprecation warning, but not the "no Int64 type" error.

Reply:
It ALSO turns out that if I changed Int64 -> uint64_t in my code, that wasn't good enough for old OpenEXR. It fails on MacOS when building against OpenEXR 2.5, because Int64 was defined as "unsigned long long" but uint64_t is "unsigned long", both of which are 64 bit on Mac, so I'm not sure why it's a conflict, but clang thinks that `long` and `long long` are always different, even on platforms where they are the same. So I ended up with a slightly more awkward

    #if [using Imath >= 3]
        blah uint64_t
    #else
       blah Imath::Int64
    #endif

Reply:
¯\_(ツ)_/¯

Reply:
Are you pointing specifically to that  #include in ImathRandom.cpp, or more generally at the fact that ImathInt64.h no longer gets included by the same other include files?  It's no longer pulled in by ImathMatrix.h, for example. We've caused that problem in a number of places, I had to add hundreds of #include <cfloat> lines to files that previously picked up FLT_MAX, etc from ImathLimits.h. Painful as that is, I still think eliminating extraneous includes is a good thing, and a major release is a good time to do it. But I could be convinced otherwise.

Reply:
Yeah, sorry, it wasn't ImathRandom.cpp in particular, it's probably the change in ImathFun.h that is my culprit.

Reply:
I left my comment in the wrong place. I just meant that generically, someplace or several places that used to include ImathInt64.h now no longer do so, bypassing straight to cstdint, and so the Int64 disappeared from the POV of my build, so I wasn't even getting the deprecation warnings, just "unknown type." Sorry for pointing to the wrong place.

Reply:
@cary-ilm , I've already posted a PR that works around this in my code, so I think if you just want to eliminate the unnecessary include file for 3.0, that's fine with me. I was just concerned that you may have inadvertently sabotaged the intent to have users encounter only the deprecation warning. 

Just add this nugget to the porting tips guide -- if you see errors about no Int64, the quick fix is to `#include <ImathInt64.h>`.